src/doc/trpl/for-loops.md

```diff
@@ -41,3 +41,44 @@ so our loop will print `0` through `9`, not `10`.
 Rust does not have the “C-style” `for` loop on purpose. Manually controlling
 each element of the loop is complicated and error prone, even for experienced C
 developers.
+
+# Enumerate
+
+When you need to keep track of how many times you already looped, you can use the `.enumerate()` function.
+
+## On ranges:
+
+```rust
+for (i,j) in (5..10).enumerate() {
+    println!("i = {} and j = {}", i, j);
+}
+```
+
+Outputs:
+
+```rust
+i = 0 and j = 5
```

Comment:
This doesn't look like Rust source code, you should probably just mark it as `text`. (Same for the output listing below.)